distributed/cli/dask_worker.py

```diff
@@ -210,6 +210,11 @@
     help="Whether or not to restart the worker after the lifetime lapses. "
     "This assumes that you are using the --lifetime and --nanny keywords",
 )
+@click.option(
+    "--drain/--no-drain",
+    default=False,
+    help="Let the worker finish its current work before closing [default: --no-drain]",
+)
```

Comment:
I like the idea of having a flag you can set which allows workers to complete current tasks without accepting new ones. It's probably most useful when tasks a very large, which is not best practice, but common enough that we should make QoL improvements for those users.

I'm not sure on the `drain` terminology. In SLURM it means the same thing as it means here, finish existing work without accepting new work. However, in Kubernetes terminology it means evict all current work and reschedule tasks on another worker. So for Dask users in the Kubernetes community this may be confusing.

Perhaps this option could be made mode explicit like `--complete-tasks-on-shutdown`. Although that's pretty long, so if you can think of something better than that's great!